cmd/plugins.go

```diff
@@ -0,0 +1,543 @@
+				}
+			}
+		}
+		if hasNewPlugin {
+			log.Infof(fmt.Sprintf("run `%s` to import the missing plugins!", log.Au().Magenta("hover plugins get")))
+		}
+	},
+}
+
+var pluginTidyCmd = &cobra.Command{
+	Use:   "tidy",
+	Short: "Removes unused platform plugins.",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) > 0 {
+			return errors.New("does not take arguments")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		assertInFlutterProject()
+		assertHoverInitialized()
+
+		desktopCmdPath := filepath.Join(buildPath, "cmd")
+		dependencyList, err := listPlatformPlugin()
+		if err != nil {
+			log.Errorf("%v", err)
+			os.Exit(1)
+		}
+
+		importedPlugins, err := ioutil.ReadDir(desktopCmdPath)
+		if err != nil {
+			log.Errorf("Failed to search for plugins: %v", err)
+			os.Exit(1)
+		}
+
+		for _, f := range importedPlugins {
+			isPlugin := strings.HasPrefix(f.Name(), "import-")
+			isPlugin = isPlugin && strings.HasSuffix(f.Name(), "-plugin.go")
+
+			if isPlugin {
+				pluginName := strings.TrimPrefix(f.Name(), "import-")
+				pluginName = strings.TrimSuffix(pluginName, "-plugin.go")
+
+				pluginInUse := false
+				for _, dep := range dependencyList {
+					if dep.name == pluginName {
+						// plugin in pubspec.lock
+						pluginInUse = true
+						break
+					}
+				}
+
+				if !pluginInUse || tidyPurge {
+					if dryRun {
+						fmt.Printf("       plugin: [%s] can be removed\n", pluginName)
+						continue
+					}
+					pluginImportPath := filepath.Join(desktopCmdPath, f.Name())
+
+					// clean-up go.mod
+					pluginImportStr, _ := readPluginGoImport(pluginImportPath, pluginName)
+					// Delete the 'replace' and 'require' import strings from go.mod.
+					// Not mission critical, if the plugins not correctly removed from
+					// the go.mod file, the project still works and the plugin is
+					// successfully removed from the flutter.Application.
+					if err != nil || pluginImportStr == "" {
+						log.Warnf("Couldn't clean the '%s' plugin from the 'go.mod' file. Error: %v", pluginName, err)
+					} else {
+						fileutils.RemoveLinesFromFile(filepath.Join(buildPath, "go.mod"), pluginImportStr)
+					}
+
+					// remove import file
+					err = os.Remove(pluginImportPath)
+					if err != nil {
+						log.Warnf("Couldn't remove plugin %s: %v", pluginName, err)
+						continue
+					}
+					fmt.Printf("       plugin: [%s] removed\n", pluginName)
+				}
+			}
+		}
+	},
+}
+
+var pluginGetCmd = &cobra.Command{
+	Use:   "get",
+	Short: "Imports missing platform plugins in the application",
+	Args: func(cmd *cobra.Command, args []string) error {
+		if len(args) > 0 {
+			return errors.New("does not take arguments")
+		}
+		return nil
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		assertInFlutterProject()
+		assertHoverInitialized()
+		hoverPluginGet()
+	},
+}
+
+func hoverPluginGetDryRun() bool {
+	dryRun = true
+	defer func() {
+		dryRun = false
+	}()
+	return hoverPluginGet()
```

Comment:
What happens if dryRun was already true? It is then disabled after this function returns. That feels like unexpected behavior.

Maybe change to: ?
```
func hoverPluginGetDryRun() bool {
	originalDryRun = dryRun
	dryRun = true
	defer func() {
		dryRun = originalDryRun
	}()
	return hoverPluginGet()
}
```

Reply:
Maybe cleaner to add dryRun as an argument to the function that actually executes stuff; `hoverPluginGet(dryRun bool)`